src/watch.hpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (C) 2022, KNS Group LLC (YADRO).
+ */
+#pragma once
+
+#include <sdeventplus/event.hpp>
+#include <sdeventplus/source/event.hpp>
+#include <sdeventplus/source/io.hpp>
+
+#include <filesystem>
+#include <map>
+
+namespace fs = std::filesystem;
+
+namespace inotify
+{
+
+/**
+ * @brief Adds inotify watch on persistent files to be synced
+ */
+class Watch
+{
+  public:
+    using Callback = std::function<void(int, const fs::path&)>;
+
+    Watch() = delete;
+    Watch(const Watch&) = delete;
+    Watch& operator=(const Watch&) = delete;
+    Watch(Watch&&) = delete;
+    Watch& operator=(Watch&&) = delete;
+
+    /**
+     * @brief dtor - remove inotify watch and close fd's
+     */
+    ~Watch();
+
+    static Watch create(sdeventplus::Event& event, const fs::path& root,
+                        Callback callback);
+
+  protected:
+    /**
+     * @brief ctor - hook inotify watch with sd-event
+     *
+     * @param event - sd-event object
+     * @param fd    - inotify fd
+     * @param root  - root directory watched to
+     * @param callback - The callback function for processing files
```

Comment:
Strange alignment. Please either align them all or don't align them all, not this strange mix. :)

Reply:
fixed